test/unit/providers/simulation-provider.test.ts

```diff
@@ -99,13 +97,99 @@ describe('fallback tenderly simulator', () => {
       async () => false
     );
     const swapRoute = await simulator.simulate(
-      fromAddressMock,
-      swapOptionsMock,
-      swapRouteMock,
-      amountMock,
-      quoteMock
+      fromAddress,
+      swapOptions,
+      swaproute,
+      amount,
+      quote
     );
     expect(simulateTxStub.called).toBeFalsy();
     expect(swapRoute.simulationStatus).toEqual(SimulationStatus.Unattempted);
   });
 });
+describe('Eth estimate gas simulator', () => {
+    const chainId = ChainId.MAINNET;
+    const simulator = new EthEstimateGasSimulator(chainId, provider, v2PoolProvider, v3PoolProvider);
+    let ethEstimateGasStub: sinon.SinonStub;
+    let simulateTxStub: sinon.SinonStub;
+
+    const swaproute: SwapRoute = {
+        quote: quote,
+        quoteGasAdjusted: quote,
+        estimatedGasUsed: estimatedGasUsed,
+        estimatedGasUsedQuoteToken: quote,
+        estimatedGasUsedUSD: quote,
+        gasPriceWei: estimatedGasUsed,
+        trade: trade,
+        route: route,
+        blockNumber: blockNumber,
+        simulationStatus: SimulationStatus.Succeeded,
+    };
+
+    beforeEach(() => {
+        simulateTxStub = sinon.stub(simulator, <any>'simulateTransaction');
+        simulateTxStub.resolves(swaproute);
+        ethEstimateGasStub = sinon.stub(simulator, <any>'ethEstimateGas');
+        ethEstimateGasStub.resolves(swaproute)
+    });
```

Comment:
These unit tests don't really make sense- you've mocked out the logic from the class that we are testing.. so what logic is actually being tested at all?

Reply:
fixed the tests